CONTRIBUTING.md

```diff
@@ -1,136 +1,228 @@
-ideal report should include the steps to reproduce it.
+If you have an idea for a new feature please open an **Enhancement** issue in
+the [issue tracker](https://github.com/Qiskit/qiskit-terra/issues). This will
+enable discus the new feature, whether it's a good fit for the project, how we
+can implement it, etc.
 
-### Doubts solving
+### Code Review
 
-To help less advanced users is another wonderful way to start. You can
-help us close some opened issues. This kind of tickets should be labeled
-as `question`.
+Code review is done in the open and open to anyone. While only maintainers have
+access to merge commits, providing feedback on pull requests is very valuable
+and helpful. It is also a good mechanism to learn about the code base. You can
+view a list of all open pull requests here:
+https://github.com/Qiskit/qiskit-terra/pulls
+to review any open pull requests and provide feedback on it.
 
-### Improvement proposal
+### Good first contributions
+
+If you would like to contribute to the qiskit-terra but aren't sure of where
+to get started, the
+[`good first issue`](
+https://github.com/Qiskit/qiskit-terra/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
+label in the issues is a good place to start. These are all issues that have
+been reviewed by contributors and tagged as something a new contributor
+should be able to develop a fix for. In other words it shouldn't require
+intimate familiarity with qiskit-terra to develop a fix for the issue.
+
+### Documentation
+
+If you make a change, make sure you update the associated
+*docstrings* and parts of the
+[documentation](https://github.com/Qiskit/qiskit/tree/master/docs/terra)
+that corresponds to it.
+
+### Pull requests
 
-If you have an idea for a new feature please open a ticket labeled as
-`enhancement`. If you could also add a piece of code with the idea or a
-partial implementation it would be awesome.
+We use [GitHub pull requests](
+https://help.github.com/articles/about-pull-requests) to accept the contributions.
+
+While not required, it is often good practice to open a new issue about
+the bug you're fixing or the feature you're implementing prior to creating
+a pull request. This will enable a discussion about what is being implemented
+and how best we can work together to implement it in the code.
+
+If you've written some code but need help finishing it, want to get initial
+feedback on it priort to finishing it, or want to share it and discuss prior
+to finishing the implementation you can open a *Work in Progress* pull request.
+When you create the pull request prefix the title with the **\[WIP\]** tag (for
+work in progress). This will indicate to reviewers that the code in the PR
+isn't in it's final state and will change. It also means that we will not merge
+the commit until it is finished. You or a reviewer can remove the [WIP] tag when
+the code is ready to fully reviewed for merging.
 
```

Comment:
"the code is ready to fully" -> "the code is ready to be fully"

Reply:
Done